modules/broker-core/src/main/java/org/wso2/broker/core/Broker.java

```diff
@@ -105,6 +114,14 @@ public void acknowledge(String queueName, Message message) throws BrokerExceptio
         metricManager.markAcknowledge();
     }
 
+    public Set<QueueHandler> prepareEnqueue(Xid xid, Message message) throws BrokerException {
```

Comment:
Shouldn't these be methods of transaction object?

Reply:
We need to consolidate Broker and MessagingEngine objects. Until then this extra hop will be there.